subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -161,63 +167,119 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec {
-                firstInput = new NestedBeanWithInput(firstInput: project.findProperty('firstInput'))
+                def firstString = project.findProperty('firstInput')
+                firstInput = new NestedBeanWithInput(firstInput: firstString, firstOutputFile: file("${firstOutputFile}"), firstInputFile: file("${firstInputFile}"))
                 outputFile.set(project.layout.buildDirectory.file('output.txt'))
             }
             
             task configureTask {
+                def secondString = project.findProperty('secondInput')
                 doLast {
-                    taskWithNestedProperty.bean = new NestedBeanWithOtherInput(secondInput: project.findProperty('secondInput'))
+                    taskWithNestedProperty.bean = new NestedBeanWithOtherInput(secondInput: secondString, secondOutputFile: file("${secondOutputFile}"), secondInputFile: file("${secondInputFile}"))
                 }
             }
             
             taskWithNestedProperty.dependsOn(configureTask)
         """
 
         def task = ':taskWithNestedProperty'
+        file(firstInputFile).text = "first input file"
+        file(secondInputFile).text = "second input file"
+
+        def inputProperties = [
+                first: 'first',
+                second: 'second'
+        ]
+        def inputFiles = [
+                first: file(firstInputFile),
+                second: file(secondInputFile)
+        ]
+        def outputFiles = [
+                first: file(firstOutputFile),
+                second: file(secondOutputFile)
+        ]
+
+        def changes = [
+                inputProperty: { String property ->
+                    inputProperties[property] = inputProperties[property] + ' changed'
+                },
+                inputFile: { String property ->
+                    inputFiles[property] << ' changed'
+                },
+                outputFile: { String property ->
+                    outputFiles[property] << ' changed'
+                }
+        ]
+
+        def runTask = { ->
+            run task, '-PfirstInput=' + inputProperties.first, '-PsecondInput=' + inputProperties.second
+        }
+
         when:
-        run task, '-PfirstInput=first', '-PsecondInput=second'
+        runTask()
 
         then:
         executedAndNotSkipped(task)
-        def outputFile = file('build/output.txt')
-        outputFile.text == 'second'
 
         when:
-        run task, '-PfirstInput=different', '-PsecondInput=second'
+        changes[change]('first')
+        runTask()
 
         then:
         skipped(task)
-        outputFile.text == 'second'
 
         when:
-        run task, '-PfirstInput=different', '-PsecondInput=secondSecond'
+        changes[change]('second')
+        runTask()
 
         then:
         executedAndNotSkipped(task)
-        outputFile.text == 'secondSecond'
+
+        where:
+        change << ['inputProperty', 'inputFile', 'outputFile']
```

Comment:
This test is now rather big and complex. Can we split it up instead?

Reply:
It might be okay to test changing everything in one go, too.

Reply:
I extracted the logic into a test fixture.
I think the test is clearer now, and the fixture is also used for the test when changing the task in question started executing. WDYT?